### ISSUE TITLE
Modifier key profile condition

### ISSUE THREAD

**Issue by siikamiika** [FooSoft/yomichan#412]
one other possible solution that came to my mind is to have different modifier keys trigger different profiles.(a new type of profile usage condition)
I don't know if this is easier to implement than the original request though.

_Originally posted by @Pintliz in https://github.com/FooSoft/yomichan/issues/320#issuecomment-574220642_

**Comment by siikamiika** [comment]
Depends on https://github.com/FooSoft/yomichan/issues/452

**Comment by siikamiika** [comment]
@toasted-nutbread I have a POC here: https://github.com/siikamiika/yomichan/commits/scanning-key-combinations-2

Any suggestions how to change the input type of the condition when changing between modifier selection and others? I created these templates first
```html
<template id="condition-input-text-template"><input type="text" class="form-control" /></template>
<template id="condition-input-select-template"><select class="form-control"></select></template>
```

but it would have been a bit complicated with the listeners. Another option would be to make the input listen to key events and choose the modifier based on the pressed key.

**Comment by toasted-nutbread** [comment]
You could use `EventListenerCollection` so you don't have to manually track all the events which need to be removed. I guess it doesn't support jQuery's `on`/`off` style currently, but you can address that by using the native element or making `EventListenerCollection` support `on`/`off`. (Aside: moving away from jQuery/Bootstrap on the settings page in the v2 updates.)

If you're intending on supporting modifier combinations (e.g. `alt + shift`), then you probably want to use the text input + detect key strokes. `<select>` can do multiple, but it's layout is different. Make sure to use esc/backspace to clear. I think this will just have to be handled as a special case in the UI for now.

**Comment by siikamiika** [comment]
Ok, thanks for the tips. I'll try what works the best. `<select>` support could be useful for other conditions in the future so I'll look at the `EventListenerCollection` approach first.

It's possible to support combinations with simple conditions too using `Modifier Key = alt AND Modifier Key = shift`. It's more complicated to implement stuff like `Modifier Key != alt + shift` on single row but might be possible with transforms.

**Comment by siikamiika** [comment]
I had to do some refactoring anyway for `select` support so I was able to contain the event listener activation within a function that creates the `<input>` or `<select>` element. This doesn't use `EventListenerCollection`: https://github.com/siikamiika/yomichan/commit/6024e4981aea7df083ace4633c2ba85c32572a05

I also realized that you probably meant exact modifier matches when you said combinations. I renamed `=` and `!=` to `has` and `doesn't have`, and I'll try to add `is` and `is not` next.
